dask/blockwise.py

```diff
@@ -383,6 +456,77 @@ def map_tasks(self, func):
         return ret
 
 
+def _get_coord_mapping(
```

Comment:
Great idea of separating the abstract coordinates from the task generation. Since `_get_coord_mapping()` implements the bulk of the blockwise complexity, I would be great if you could add some doc describing the input/output arguments and how the coordinates are calculated.

Reply:
Right - That makes sense :)

I added a simple docstring with more info, but I may need to expand a bit further.